cmd/curio/guidedsetup/guidedsetup.go

```diff
@@ -783,16 +785,46 @@ func stepNewMinerConfig(d *MigrationData) {
 			d.say(notice, "Please do not run guided-setup again as miner creation is not idempotent. You need to run 'curio config new-cluster %s' to finish the configuration", d.MinerID.String())
 			os.Exit(1)
 		}
+		stepCompleted(d, d.T("Configuration 'base' was updated to include this miner's address"))
+		return
 	}
 
-	_, err = d.DB.Exec(d.ctx, "INSERT INTO harmony_config (title, config) VALUES ($1, $2)", d.MinerID.String(), configTOML.String())
+	baseCfg := &config.CurioConfig{}
+	baseText := ""
+
+	err = d.DB.QueryRow(d.ctx, "SELECT config FROM harmony_config WHERE title='base'").Scan(&baseText)
+	if err != nil {
+		d.say(notice, "Cannot load base config: %s", err.Error())
+		d.say(notice, "Please do not run guided-setup again as miner creation is not idempotent. You need to run 'curio config new-cluster %s' to finish the configuration", d.MinerID.String())
+		os.Exit(1)
+	}
+	EnsureEmptyArrays(baseCfg)
```

Comment:
no good. If they're multi-miner, this would zero-out their first miner ID.
Lets preserve pre-existing address for the multi-miner case.

Reply:
Yes, I noticed that. I will be fix all the current issues and test it again in devnet.